### PR TITLE
Let Guava handle the Connection Throttling

### PR DESCRIPTION
**Pull request by PunKeel**
Less code, less bugs.

The only valid reason I see for this change is : The test failed on my computer, and I don't understand why the code has to be so _ugly_.

### CODE REVIEW

File: proxy/src/main/java/net/md_5/bungee/ConnectionThrottle.java [hunk 1]
```diff
@@ -8,25 +8,25 @@
 public class ConnectionThrottle
 {
 
-    private final int throttleTime;
     private final Cache<InetAddress, Long> throttle;
```
Comment (minecrafter) [inline proxy/src/main/java/net/md_5/bungee/ConnectionThrottle.java]: Needs to be `Cache<InetAddress, Boolean>`

Comment (PunKeel) [inline proxy/src/main/java/net/md_5/bungee/ConnectionThrottle.java]: Indeed. Forgotten when I made the patch here

**Comment by Janmm14** [comment]
@PunKeel Yes, we can let guava handle the expiration.

Also the test is not failing in the official build and my private build and on travis.

Your code contains a bug and an unintended change.
 
You are only adding an ip to the map if the ip is in the map. Your if-check needs to be reverted.

If that is changed, the function of the throttle however has changed by you changing ```expireAfterAccess``` to ```expireAfterWrite```. The throttle time should be renewed each connection attempt, ensuring a minimum delay between every connection attempt and not between every successful connection attempt.

**Comment by PunKeel** [comment]
@Janmm14: My raspberry pi is too slow. That's the issue with time dependent tests: they fail. Using Guava would allow me to use a custom Clock / Ticker, but I haven't commited it here. To be honest, this is why I made the if bug.

For the behavior change, this was intented in my modification (set a 5s throttle, without renewing it everytime a throttle occurs) but is unwanted here.

**Comment by minecrafter** [comment]
@PunKeel: If that's the reason why you're committing this, then we should probably increase the intervals instead, solving the immediate issue. Instead of 15 milliseconds, would 200 work?

**Comment by Janmm14** [comment]
I do not like this behaviour change.

**Comment by PunKeel** [comment]
@minecrafter: I haven't found a clean way to provide the Clock to Guava, as it must be given to the CacheBuilder...

@Janmm14: that's why I reverted the behavior to the old one. :-)

**Comment by md-5** [comment]
d9a8311b8e40a0d390a993e409933f8ee737c12e

**Pull request closed without merging**